src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -88,6 +89,13 @@ public class PhysicalFilesWatcher : IDisposable
 
             if (fileSystemWatcher != null)
             {
+#if NETCOREAPP
+                if (OperatingSystem.IsBrowser())
+                {
+                    throw new PlatformNotSupportedException(SR.Format(SR.FileSystemWatcher_PlatformNotSupported, typeof(FileSystemWatcher)));
```

Comment:
Same question here, if it throws for browser all time it should have `[UnsupportedOSPlatform("browser")]`?

Reply:
Because it doesn't throw for browser all the time; it only throws if `fileSystemWatcher` is not null.

Reply:
OK

Reply:
Is there a way to signal that something is not supported if certain parameter is null/not null?

Reply:
No, we have been talking about that, but there were not many (known) cases to add such option

Reply:
Well, I think this is one case 😃.

Reply:
yeah, we might add that in the future